src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/CSharpFormattingPassBase.cs

```diff
@@ -379,10 +379,21 @@ bool IsImplicitOrExplicitExpression()
                 return owner.AncestorsAndSelf().Any(n => n is CSharpImplicitExpressionSyntax || n is CSharpExplicitExpressionSyntax);
             }
 
-            bool IsInSectionDirective()
+            bool IsInSectionDirectiveCloseBrace()
             {
-                var directive = owner.FirstAncestorOrSelf<RazorDirectiveSyntax>();
-                if (directive != null &&
+                // @section Scripts {
+                //     <script></script>
+                // }
+                //
+                // We are fine to format these, but due to how they are generated (inside a multi-line lambda)
+                // we want to exlude the final close brace from being formatted, or it will be indented by one
+                // level due to the lambda. The rest we don't need to worry about, because the one level indent
+                // is actually desirable.
+                if (owner is MarkupTextLiteralSyntax &&
+                    owner.Parent is MarkupBlockSyntax block &&
+                    owner == block.Children[block.Children.Count - 1] &&
+                    // MarkupBlock -> CSharpCodeBlock -> RazorDirectiveBody -> RazorDirective
+                    block.Parent.Parent.Parent is RazorDirectiveSyntax directive &&
```

Comment:
Lol damn, I would have figured this would have null-ref'd at the top level document for normal HTML but if that were the case our tests would be on the floor. /cc @javiercn for another example of sadness w/ the SyntaxTree

Reply:
I was surprised at this too, but all of the formatting code is null-annotated, so I'm trusting the compiler that there are no warnings here. Happy to add some question marks if you don't trust the compiler though :)